compiler/rustc_codegen_llvm/src/debuginfo/metadata.rs

```diff
@@ -636,6 +636,7 @@ pub fn file_metadata<'ll>(cx: &CodegenCx<'ll, '_>, source_file: &SourceFile) ->
             rustc_span::SourceFileHashAlgorithm::Md5 => llvm::ChecksumKind::MD5,
             rustc_span::SourceFileHashAlgorithm::Sha1 => llvm::ChecksumKind::SHA1,
             rustc_span::SourceFileHashAlgorithm::Sha256 => llvm::ChecksumKind::SHA256,
+            rustc_span::SourceFileHashAlgorithm::Blake3 => llvm::ChecksumKind::None,
```

Comment:
This would effectively disable support for the check of debuggers that the source files are unchanged since compilation once cargo starts using file checksums, which can cause confusion when debugging. Can we use a separate hash to encode in the debuginfo and to store in the dep info file?

Reply:
This PR already does that. I didn’t want the two hash algorithms to be bound like that.